jwql/instrument_monitors/common_monitors/readnoise_monitor.py

```diff
@@ -342,28 +342,34 @@ def make_readnoise_image(self, data):
             The 2D readnoise image.
         """
 
-        # Create a stack of correlated double sampling (CDS) images using input
-        # ramp data, combining multiple integrations if necessary.
-        logging.info('\tCreating stack of CDS difference frames')
+        logging.info('\tCreating readnoise image')
         num_ints, num_groups, num_y, num_x = data.shape
-        for integration in range(num_ints):
-            if num_groups % 2 == 0:
-                cds = data[integration, 1::2, :, :] - data[integration, ::2, :, :]
-            else:
-                # Omit the last group if the number of groups is odd
-                cds = data[integration, 1::2, :, :] - data[integration, ::2, :, :][:-1]
 
-            if integration == 0:
-                cds_stack = cds
-            else:
-                cds_stack = np.concatenate((cds_stack, cds), axis=0)
+        # Calculate the readnoise in slices to avoid memory issues on large files.
+        slice_width = 20
```

Comment:
Nice addition here, I this should help with the memory pressure we have been experiencing.

Reply:
thanks yeah it used a lot less when I was testing locally